figures/tasks.py

```diff
@@ -128,8 +129,9 @@ def populate_daily_metrics(date_for=None, force_update=False):
     logger.info('Starting task "figures.populate_daily_metrics" for date "{}"'.format(
         date_for))
 
-    sites_count = Site.objects.using(read_replica_or_default()).count()
-    for i, site in enumerate(Site.objects.using(read_replica_or_default()).all()):
+    lms_sites = EdlySubOrganization.objects.using(read_replica_or_default()).values_list('lms_site')
+    sites_count = len(lms_sites)
+    for i, site in enumerate(Site.objects.using(read_replica_or_default()).filter(id__in=lms_sites)):
```

Comment:
@asadmanzoor93  could you please try this `Site.objects.using(read_replica_or_default()).filter(lms_site__isnull=Fasle)`

Reply:
@burhandodhy `lms_site` is not a field in Site and we get the below error. 
<img width="1123" alt="Screenshot 2021-05-05 at 2 59 02 PM" src="https://user-images.githubusercontent.com/17013371/117124997-709c2f80-adb2-11eb-9599-fdbb5a31d354.png">

Reply:
Applogy please try this `Site.objects.using(read_replica_or_default()).filter(edly_sub_org_for_lms__isnull=False)` it will only return `lms` sites

Reply:
For `site_count` we have to make a query in either way. So utilizing that we directly get the lms sites associated from `EdlySubOrganization` model instead of reverse checking again in for loop for the site.